### PR TITLE
Implemented gather_nd frontend function

### LINKED ISSUE #20825: gather_nd

### PR DESCRIPTION
**Pull request by aazam-gh**
Closes #20825

### CODE REVIEW

**Comment by aazam-gh** [comment]
There's a bit of work to be done in having the tests pass. The `index` parameter needs to satisfy the condition described in the paddle docs `index (Tensor) – The index input with rank > 1, index.shape[-1] <= input.rank. Its dtype should be int32, int64.index (Tensor) – The index input with rank > 1, index.shape[-1] <= input.rank. Its dtype should be int32, int64.`

**Review by kurshakuz — CHANGES_REQUESTED** [review]
Can you please resolve conflicts and show the results of pytest for your implemenation? Thanks!

**Comment by aazam-gh** [comment]
@kurshakuz I have resolved the conflicts and here's a screenshot of the results
image

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_manipulation.py [hunk 1]
```diff
@@ -601,3 +601,47 @@ def test_paddle_roll(
...
     )
+
+
+#gather_nd
+@st.composite
+def _input_and_index(draw):
+    dtype = draw(helpers.get_dtypes("float", full=False))
+    dtype_and_input = draw(helpers.dtype_and_values(dtype=dtype, min_num_dims=1))  # ensure at least one dimension
+    print(dtype_and_input)
+    params_rank = len(dtype_and_input[1].shape)
```
Comment (kurshakuz) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_manipulation.py:612]: Hi! Your implementation is failing on this line. `dtype_and_input[1]` is a list object which does not contain `shape` attribute. Please make sure that your implementation does not produce any errors and pass first. Let me once it is fixed for a final PR review. Thanks!

**Review by kurshakuz — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_manipulation.py (hunk 1 above)
Comment (aazam-gh) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_manipulation.py:612]: Thanks for the review, but can you help me out on how to alternatively make the condition pass.

**Pull request closed without merging**